src/main/java/com/google/sps/data/UserQuery.java

```diff
@@ -79,8 +71,19 @@ public UserQuery(String query, String tags, String authors) {
       );
       searchFilters.add(tempAuthorFilter);
     }
-    if (recipeName != null && !recipeName.equals("")) {
-      searchFilters.add(new Query.FilterPredicate("name", Query.FilterOperator.EQUAL, recipeName.toLowerCase().trim()));
+
+    if (tagsAndIngredients.size() > 1) {
+      searchFilters.add(createTagsFilter());
+    } else if (tagsAndIngredients.size() == 1) {
+      // We check for size equal to 1 because we cannot create a composite filter when we only 
+      // have one filter to pass. The same implementatoin is used for the author filters
+      // Use of collection interface methods only to allow for future flexibility
+      String tempTag = tagsAndIngredients.iterator().next();
+      searchFilters.add(createSingleTagFilter(tempTag));
+    }
+
+    if (!recipeName.equals("null") && !recipeName.equals("")) {
```

Comment:
Is this strange? Yes. Does it work? Yes. Did I try `recipeName != null`? Yes, and it did not work. I had another friend in STEP help me try to figure it out and our closest guess is that when the client sends a fetch request with a param with no values, it decides to send a string version of the value "null" that the server can use. That's why we get this behavior (where the first null is line 88): 
![Screenshot 2020-08-03 at 7 36 22 PM](https://user-images.githubusercontent.com/42452832/89319933-d2c2db00-d64e-11ea-9532-b58cd3836e5f.png)